cmake/Modules/MacroQbtCommonConfig.cmake

```diff
@@ -0,0 +1,117 @@
+# Set common variables and create some interface-only library targets
+# that some or all other targets will link to, either directly or transitively,
+# to consume common compile options/definitions
+
+macro(qbt_common_config)
+
+    # treat value specified by the CXX_STANDARD target property as a requirement by default
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+    # these definitions are only needed for calls to
+    # lt::generate_fingerprint and for the qbittorrent.rc file on Windows
+    add_library(qbt_version_definitions INTERFACE)
+
+    target_compile_definitions(qbt_version_definitions
+        INTERFACE
+            QBT_VERSION_MAJOR=${qBittorrent_VERSION_MAJOR}
+            QBT_VERSION_MINOR=${qBittorrent_VERSION_MINOR}
+            QBT_VERSION_BUGFIX=${qBittorrent_VERSION_PATCH}
+            QBT_VERSION_BUILD=${qBittorrent_VERSION_TWEAK}
+    )
+
+    add_library(qbt_common_cfg INTERFACE)
+
+    # Full C++ 14 support is required
+    target_compile_features(qbt_common_cfg
+        INTERFACE
+            cxx_std_14
+            cxx_contextual_conversions
+            cxx_binary_literals
+            cxx_decltype_auto
+            cxx_lambda_init_captures
+            cxx_generic_lambdas
+            cxx_variable_templates
+            cxx_relaxed_constexpr
+            cxx_attribute_deprecated
+            cxx_digit_separators
+    )
+
+    set(QBT_FULL_VERSION "${qBittorrent_VERSION}${QBT_VER_STATUS}")
+
+    target_compile_definitions(qbt_common_cfg
+        INTERFACE
+            QBT_VERSION="v${QBT_FULL_VERSION}"
+            QBT_VERSION_2="${QBT_FULL_VERSION}"
+            QT_DEPRECATED_WARNINGS
+            QT_NO_CAST_TO_ASCII
+            QT_NO_CAST_FROM_BYTEARRAY
+            QT_USE_QSTRINGBUILDER
+            QT_STRICT_ITERATORS
+            # GENEXP: the $<CONFIG:...> genexp is the only robust way of providing
+            # build type-dependent functionality for both single config and multi-config-generators alike
+            $<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG_OUTPUT>
+    )
+
+    if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        target_compile_definitions(qbt_common_cfg
+            INTERFACE
+                NTDDI_VERSION=0x06010000
+                _WIN32_WINNT=0x0601
+                _WIN32_IE=0x0601
+                UNICODE
+                _UNICODE
+                WIN32_LEAN_AND_MEAN
+                _CRT_SECURE_NO_DEPRECATE
+                _SCL_SECURE_NO_DEPRECATE
+                NOMINMAX
+        )
+    elseif (UNIX AND (NOT APPLE) AND (NOT CYGWIN))
+        target_compile_definitions(qbt_common_cfg INTERFACE _DEFAULT_SOURCE)
```

Comment:
@glassez 

I mostly left it there because it was there previously. It's a glibc thing, I think it's "common practice" to define it. Otherwise, gcc can complain about some implicit function declartions, if we use those certain functions.

There is probably no problem in defining it in other systems I guess (they probably just ignore it). I wonder if it's a problem for Linux systems using something like `musl` though.

At the moment, we don't need it - I just tried building without it and got no warnings. Should I remove it? I think this is mostly for C stuff anyway, not C++.

More info: 

https://stackoverflow.com/questions/29201515/what-does-d-default-source-do (also read the second answer)
https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html.

Reply:
At least you shouldn't separate different Unix systems according to it. Let either declare or remove it everywhere.

Reply:
@glassez Well, since you have no reasons for keeping it either, I've removed it.

Reply:
>@glassez Well, since you have no reasons for keeping it either, I've removed it.

Don't count on me here. I'm not sure what it is, why it's needed, or whether we can safely remove it.

Reply:
@glassez To the best of my knowledge right now, I think it is safe to remove it.